firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionNavigationHeaderView.swift

```diff
@@ -0,0 +1,105 @@
+import UIKit
+import Common
+import Shared
+import ComponentLibrary
+
+class TrackingProtectionNavigationHeaderView: UIView {
+    private struct UX {
+        static let closeButtonSize: CGFloat = 30
+        static let imageMargins: CGFloat = 10
+        static let baseDistance: CGFloat = 20
+    }
+
+    let siteTitleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .center
+        label.font = FXFontStyles.Regular.headline.scaledFont()
+        label.numberOfLines = 2
+        label.accessibilityTraits.insert(.header)
+    }
+
+    private var closeButton: CloseButton = .build { button in
+        button.layer.cornerRadius = 0.5 * UX.closeButtonSize
+    }
+
+    var backButton: UIButton = .build { button in
+        button.layer.cornerRadius = 0.5 * UX.closeButtonSize
+        button.clipsToBounds = true
+        button.setTitle(.KeyboardShortcuts.Back, for: .normal)
+        button.setImage(UIImage(imageLiteralResourceName: StandardImageIdentifiers.Large.chevronLeft)
+            .withRenderingMode(.alwaysTemplate),
+                        for: .normal)
+        button.titleLabel?.font = FXFontStyles.Regular.headline.scaledFont()
+    }
+
+    private let horizontalLine: UIView = .build { _ in }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: View Setup
+    private func setupView() {
+        addSubviews(siteTitleLabel, backButton, closeButton, horizontalLine)
+
+        NSLayoutConstraint.activate([
+            backButton.leadingAnchor.constraint(
+                equalTo: leadingAnchor,
+                constant: UX.imageMargins
+            ),
+            backButton.centerYAnchor.constraint(equalTo: centerYAnchor),
+
+            siteTitleLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
+            siteTitleLabel.leadingAnchor.constraint(equalTo: backButton.trailingAnchor),
+            siteTitleLabel.trailingAnchor.constraint(equalTo: closeButton.leadingAnchor),
+            siteTitleLabel.topAnchor.constraint(
+                equalTo: topAnchor,
+                constant: UX.baseDistance
+            ),
+            siteTitleLabel.bottomAnchor.constraint(
+                equalTo: bottomAnchor,
+                constant: -UX.baseDistance
+            ),
+
+            closeButton.trailingAnchor.constraint(
+                equalTo: trailingAnchor,
+                constant: -TPMenuUX.UX.horizontalMargin
+            ),
+            closeButton.centerYAnchor.constraint(equalTo: centerYAnchor),
+            closeButton.heightAnchor.constraint(greaterThanOrEqualToConstant: UX.closeButtonSize),
+            closeButton.widthAnchor.constraint(greaterThanOrEqualToConstant: UX.closeButtonSize),
+            closeButton.widthAnchor.constraint(equalTo: closeButton.heightAnchor),
```

Comment:
You may need to set the priority of these two constraints. Otherwise, if the parent view were to shrink, the constraint that breaks first will be random. 🤔  I'm guessing maybe the equalTo anchor should be one less than the default priority? Will need to test on iPhone and iPad.

Reply:
![Simulator Screenshot - iPhone 15 Pro Max - 2024-09-06 at 10 08 19](https://github.com/user-attachments/assets/8cf64036-e280-40f3-8715-f94083ae3e56)
![Simulator Screenshot - iPhone 15 Pro Max - 2024-09-06 at 10 09 02](https://github.com/user-attachments/assets/0efcc7d1-4d94-4c06-9f04-1f48ab0ed4fc)